File: core-webserver/core-entities/src/main/java/com/example/starter/db/entity/TenantEntity.java
package com.example.starter.db.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import java.util.UUID;
import org.jetbrains.annotations.Nullable;

@Entity
public class TenantEntity {

  @Id
  @GeneratedValue
  @Nullable
  public UUID id;
  @Column(nullable = false)
  public String name;

  public TenantEntity(String name) {
    this.name = name;
  }

  TenantEntity() {
  }
}

File: src/main/kotlin/com/example/starter/db/TenantEntity.kt
package com.example.starter.db

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import java.util.UUID

@Entity
class TenantEntity private constructor(
    @Id @GeneratedValue
    var id: UUID?,
    @Column(nullable = false) var name: String?,
    @Suppress("UnusedPrivateMember") unused: Void?
) {
    constructor() : this(null, null, null)
    constructor(name: String) : this(null, name, null)
}
